### PR TITLE
add support for post aggregators

### PR DESCRIPTION
**Pull request by cosmic-chichu**
- use pydruid fork with support for doublessketch
- add support for passing post aggregations

### CODE REVIEW

File: numalogic/connectors/druid.py [hunk 1]
```diff
@@ -100,14 +104,21 @@ def fetch(
         dimensions: list[str],
         granularity: str = "minute",
         aggregations: Optional[dict] = None,
+        post_aggregations: Optional[dict] = None,
         group_by: Optional[list[str]] = None,
-        pivot: Optional[Pivot] = None,
+        pivot: Optional[Pivot] = Pivot(),
```
Comment (ab93) [inline numalogic/connectors/druid.py:109]: Any reason why we are initializing here?

File: pyproject.toml [hunk 2]
```diff
@@ -43,7 +43,7 @@ pynumaflow = "~0.5"
 mlflow-skinny = { version = "^2.0", optional = true }
 redis = {extras = ["hiredis"], version = "^5.0", optional = true}
 boto3 = { version = "^1.24.64", optional = true }
-pydruid= {git="https://github.com/cosmic-chichu/pydruid.git", branch = "master", optional = true }
+pydruid= {git="https://github.com/cosmic-chichu/pydruid.git", branch = "addDoublesSketch", optional = true }
```
Comment (ab93) [inline pyproject.toml:46]: Can you push to your master branch instead?

File: tests/connectors/test_druid.py [hunk 3]
```diff
@@ -74,8 +123,13 @@ def test_build_param(self):
...
 
         filter_pairs = make_filter_pairs(["ciStatus"], ["false"])
-        actual = build_params("", "foo", ["bar"], filter_pairs, "all", float(24))
-        actual["intervals"] = ""
+        actual = build_params(
+            datasource="foo",
+            dimensions=["bar"],
+            filter_pairs=filter_pairs,
+            granularity="all",
+            hours=float(24),
```
Comment (ab93) [inline tests/connectors/test_druid.py:131]:
```suggestion
            hours=24.0,
```

**Review by ab93 — COMMENTED** [review]
Left a few comments

File: numalogic/connectors/druid.py (hunk 1 above)
Comment (cosmic-chichu) [reply numalogic/connectors/druid.py:109]: Otherwise pivot.columns throws an error in the method

File: pyproject.toml (hunk 2 above)
Comment (cosmic-chichu) [reply pyproject.toml:46]: Updating master will change the upstream PR for intervals isn't it?

File: numalogic/connectors/druid.py (hunk 1 above)
Comment (ab93) [reply numalogic/connectors/druid.py:109]: But we can't use a mutable type in the function arts. Need to find a different way here.

File: numalogic/connectors/druid.py [hunk 4]
```diff
@@ -100,14 +104,21 @@ def fetch(
         dimensions: list[str],
         granularity: str = "minute",
         aggregations: Optional[dict] = None,
+        post_aggregations: Optional[dict] = None,
```
Comment (qhuai) [inline numalogic/connectors/druid.py:107]: This comment is off the main topic: build_params() debug-logs the query params, while this fetcher init() debug-logs the query latency. Can we combine both debug logs into one log?

When logs from many vertices / pods are ingested into log store (such as Splunk), two locally back-to-back log lines may be interleaved by other log lines, making it harder to correlated separate logs.

Comment (cosmic-chichu) [reply numalogic/connectors/druid.py:103]: I've moved them into a single log

File: pyproject.toml
Comment (ab93) [inline pyproject.toml:1]: Let's remove the fork then?

File: numalogic/connectors/utils/postaggregator.py [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+
+class QuantilesDoublesSketchToQuantile(Postaggregator):
+    """Class for building QuantilesDoublesSketchToQuantile post aggregator."""
+
+    def __init__(self, field: Field, fraction: float, output_name=None):
+        if output_name is None:
+            name = "quantilesDoublesSketchToQuantile"
+        else:
+            name = output_name
```
Comment (ab93) [inline numalogic/connectors/utils/postaggregator.py:12]:
```suggestion
        name = output_name or "quantilesDoublesSketchToQuantile"
```

File: numalogic/connectors/utils/postaggregator.py [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+
+        Postaggregator.__init__(self, None, None, name)
```
Comment (ab93) [inline numalogic/connectors/utils/postaggregator.py:14]:
```suggestion
        super().__init__(None, None, name)
```

**Review by ab93 — COMMENTED** [review]
As discussed offline let's refactor the dir structure

**Pull request closed without merging**